sw/device/tests/sim_dv/pwrmgr_normal_sleep_all_wake_ups.c

```diff
@@ -0,0 +1,340 @@
+        .wakeup_src = kDifPwrmgrWakeupRequestSourceFour,
+        .config = prgm_usb_wakeup,
+    },
+    {
+        .name = "AONTIMER",
+        .dif_handle = &aon_timer,
+        .wakeup_src = kDifPwrmgrWakeupRequestSourceFive,
+        .config = prgm_aontimer_wakeup,
+    },
+    {
+        .name = "SENSOR_CTRL",
+        .dif_handle = &sensor_ctrl,
+        .wakeup_src = kDifPwrmgrWakeupRequestSourceSix,
+        .config = prgm_sensor_ctrl_wakeup,
+    },
+};
+
+/**
+ * External interrupt handler.
+ */
+void ottf_external_isr(void) {
+  dif_pwrmgr_irq_t irq_id;
+  top_earlgrey_plic_peripheral_t peripheral;
+
+  isr_testutils_pwrmgr_isr(plic_ctx, pwrmgr_isr_ctx, &peripheral, &irq_id);
+
+  // Check that both the peripheral and the irq id is correct
+  CHECK(peripheral == kTopEarlgreyPlicPeripheralPwrmgrAon,
+        "IRQ peripheral: %d is incorrect", peripheral);
+  CHECK(irq_id == kDifPwrmgrIrqWakeup, "IRQ ID: %d is incorrect", irq_id);
+}
+
+static void execute_test(uint32_t wakeup_source) {
+  // Configure wakeup device
+  kTestWakeupSources[wakeup_source].config(
+      kTestWakeupSources[wakeup_source].dif_handle);
+  // Normal sleep
+  pwrmgr_testutils_enable_low_power(
+      &pwrmgr, kTestWakeupSources[wakeup_source].wakeup_src,
+      kDifPwrmgrDomainOptionMainPowerInLowPower);
+  LOG_INFO("Issue WFI to enter sleep %d", wakeup_source);
+  wait_for_interrupt();
+}
+
+/**
+ * Clean up pwrmgr wakeup reason register for the next round
+ */
+static void delay_n_clear(uint32_t delay_in_us) {
+  busy_spin_micros(delay_in_us);
+  CHECK_DIF_OK(dif_pwrmgr_wakeup_reason_clear(&pwrmgr));
+}
+
+/**
+ * Clean up wakup sources
+ */
+static void cleanup(uint32_t test_idx) {
+  switch (test_idx) {
+    case PWRMGR_PARAM_SYSRST_CTRL_AON_WKUP_REQ_IDX:
+      CHECK_DIF_OK(dif_sysrst_ctrl_ulp_wakeup_clear_status(&sysrst_ctrl));
+      break;
+    case PWRMGR_PARAM_ADC_CTRL_AON_WKUP_REQ_IDX:
+      CHECK_DIF_OK(dif_adc_ctrl_filter_match_wakeup_set_enabled(
+          &adc_ctrl, kDifAdcCtrlFilter5, kDifToggleDisabled));
+      busy_spin_micros(70);
+      break;
+    case PWRMGR_PARAM_PINMUX_AON_PIN_WKUP_REQ_IDX:
+      CHECK_DIF_OK(dif_pinmux_wakeup_cause_clear(&pinmux));
+      break;
+    case PWRMGR_PARAM_PINMUX_AON_USB_WKUP_REQ_IDX:
+      usbdev_set_wake_module_active(false);
+      CHECK_DIF_OK(dif_pinmux_wakeup_cause_clear(&pinmux));
+      break;
+    case PWRMGR_PARAM_AON_TIMER_AON_WKUP_REQ_IDX:
+      CHECK_DIF_OK(dif_aon_timer_wakeup_stop(&aon_timer));
+      CHECK_DIF_OK(dif_aon_timer_clear_wakeup_cause(&aon_timer));
+      break;
+    case PWRMGR_PARAM_SENSOR_CTRL_WKUP_REQ_IDX:
+      // clear event trigger
+      CHECK_DIF_OK(dif_sensor_ctrl_set_ast_event_trigger(&sensor_ctrl, 0,
+                                                         kDifToggleDisabled));
+      CHECK_DIF_OK(dif_sensor_ctrl_clear_recov_event(&sensor_ctrl, 0));
+      break;
+    default:
+      LOG_ERROR("unknown test index %d", test_idx);
+  }
+  delay_n_clear(30);
```

Comment:
@jdonjdon can you explain 30 us between each wake-up?

Reply:
Add to comment in the file.
at the end of each round, checker expect only one "wakeup reason" to be set.
To do that, current wakeup reason has to be cleaned before a new test starts.
And 30us is picked such as the maximum of all test cases except AON_WKUP, which
needs much longer  than others

Reply:
Thanks @jdonjdon